### ISSUE TITLE
Proposal: Allow customization of binding address for pyarmor-auth via `--host` flag 

### ISSUE THREAD

**Issue by gooraje** [dashingsoft/pyarmor#1773]
# Use case and justification

In some cases setting the binding address of `pyarmor-auth` to `0.0.0.0` may conflict with security policies.
This is currently preventing us from fully utilizing the Docker capabilities of Pyarmor within our particular build environment.
Allowing the user to control the binding address through a command line option can circumvent this issue.
We think this change could benefit others as well. @jondy would you be willing to add this to the codebase?

# Proposed changes
Change:
https://github.com/dashingsoft/pyarmor/blob/6e196025cc5b806e73bc1592dd65cd5088f60229/src/cli/docker.py#L34-L39
To:
```python
CONFIG = {
    'host': '0.0.0.0',
    'port': 29092,
    'home': os.path.expanduser(os.path.join('~', '.pyarmor', 'docker')),
    'machid': None,
    'ctx': None,
}
```
To preserve defaults.

Change:
https://github.com/dashingsoft/pyarmor/blob/6e196025cc5b806e73bc1592dd65cd5088f60229/src/cli/docker.py#L109-L120
To:
```python
def main_entry():
    parser = argparse.ArgumentParser()
    parser.add_argument('-d', '--debug', action='store_true',
                        help='Enable debug mode')
    parser.add_argument('--host', type=str, default=CONFIG['host'])
    parser.add_argument('-p', '--port', type=int, default=CONFIG['port'],
                        help=argparse.SUPPRESS)
    parser.add_argument('-s', '--sock', default='/var/run/docker.sock',
                        help=argparse.SUPPRESS)
    parser.add_argument('--home', help=argparse.SUPPRESS)
    parser.add_argument('regfile', nargs=1,
                        help='group device registration file for this machine')
    args = parser.parse_args(sys.argv[1:])
```

Finally change:
https://github.com/dashingsoft/pyarmor/blob/6e196025cc5b806e73bc1592dd65cd5088f60229/src/cli/docker.py#L138
To:
```python
host, port = args.host, args.port
```

**Comment by jondy** [comment]
No problem. This patch will be merged in next release.

**Comment by jondy** [comment]
Next version v8.5.9 will be released about in 2 weeks (Jun. 1)

**Comment by jondy** [comment]
Fixed in v8.5.9
